authprogs/authprogs.py

```diff
@@ -188,14 +189,24 @@ def check_client_ip(self, rule):
         allow_from = rule.get('from')
         if not isinstance(allow_from, list):
             allow_from = [allow_from]
-        client_ip = self.get_client_ip()
-
-        if client_ip in allow_from:
-            self.logdebug('client_ip %s in %s\n' % (client_ip, allow_from))
-            return True
-        else:
-            self.logdebug('client_ip %s not in %s' % (client_ip, allow_from))
-            return False
+        def ipnet(addr):
+            try:
+                return ipaddress.ip_network(addr, strict=False)
+            except ValueError:
+                return None
+        allow_from = [
+            ipnet(x.decode())
+            for x in allow_from
+        ]
+        allow_from = filter(lambda x: x, allow_from)
+        client_ip = ipaddress.ip_address(self.get_client_ip().decode())
+
+        for allow in allow_from:
+            if allow == u'*' or client_ip in allow:
```

Comment:
This wildcard doesn't work, it is clobbered by the ipnet function. 0.0.0.0/0 does work though :)

Reply:
O rite. Iteration detritus. I wanted a shorthand for "all" but the mapping should be in ipnet, not here.